File: 프로그래머스/lv1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.py
def solution(arr, divisor):
    answer = []
    
    if divisor == 1:
        answer = arr
    else:
        for num in arr:
            if num % divisor == 0:
                answer.append(num)
    if answer:
        answer.sort()
    else:
        answer = [-1]
    
    return answer

File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int divisor) {
        List<Integer> answer = new ArrayList<>();
        Arrays.sort(arr);
        
        for (int num : arr) {
            if (num % divisor == 0) {
                answer.add(num);
            }
        }
        
        if (answer.size() == 0) {
            answer.add(-1);
        }
        
        return answer.stream()
            .mapToInt(Integer::intValue)
            .toArray();
    }
}